ledger/block/src/verify.rs

```diff
@@ -388,9 +388,12 @@ impl<N: Network> Block<N> {
         )?;
 
         // Calculate the expected coinbase reward.
-        let expected_coinbase_reward = coinbase_reward(
+        let expected_coinbase_reward = coinbase_reward::<N>(
```

Comment:
nit: once `coinbase_reward` is bounded by `N: Network`, all the parameters that are derived from `N` can be omitted to reduce boilerplate (they can be obtained ad-hoc in the function body)

Reply:
I had initially thought about this as well, but refrained to keep the formula open s.t. `N: Network` being passed into `coinbase_reward` and `block_reward` is only used for `N::CONSENSUS_V2_HEIGHT`. 

This enables better testing for variables that go into the actual calculation of the rewards rather than being opinionated on fixed trait values.